artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -1062,6 +1062,9 @@ public final Object getDuplicateProperty() {
 
    @Override
    public boolean isDurable() {
+      if (!isLargeMessage()) {
+         ensureMessageDataScanned();
+      }
```

Comment:
I'm still working the details on this part. 
if you called isDurable at this point, it would return false. And that's having some implications.

Reply:
if you call isDurable right after reload or initial parsing, it will return false when is in fact durable.

that would have certain implications on the mirror as I need that earlier on.

I'm still figuring out what's the right thing I should do here.. It seems a bug to me on isDurable() or on the initial parsing.

Reply:
It certainly seems reasonable that isDurable should return the right result. I dont think calling isDurable should have any significant side effects normally though. I see the 'large message' persister persists whether the message is durable or not and look at that first...could the 'not-large' messages do the same?

If not that, specifically parsing just the header (if present) at startup would seem the more obvious approach than isDurable needing to do stuff.

Reply:
if you need isDurable before a full parse happened, it's always returning false at the moment.

I'm still debugging this area here.

Reply:
Yeah..so perhaps an intermediate 'scan header' could be useful for startup usage. Or the equivalent of what the large-message does, and storing the durability flag for observation on load (as happens for the expiration).

Reply:
@gemmellr right.. that's what I'm aiming to.. some sort of scan header..

The fix I have in place is just so I can continue in other changes before I can come back into this one.

I will probably need to raise another JIRA on this one.. as it seems a bug to me.

Reply:
This is being handled at #3650


Will mark this as resolved, please don't comment on the resolved conversation